python/PiFinder/main.py

```diff
@@ -498,29 +498,50 @@ def main(script_name=None, has_server=False):
                                 debug_dt = shared_state.datetime()
 
                                 # write images
-                                debug_image.save(f"{test_image_path}/{uid}_raw.png")
+                                debug_image.save(
+                                    f"{utils.debug_dump_dir}/{uid}_raw.png"
+                                )
                                 debug_image = subtract_background(debug_image)
                                 debug_image = debug_image.convert("RGB")
                                 debug_image = ImageOps.autocontrast(debug_image)
-                                debug_image.save(f"{test_image_path}/{uid}_sub.png")
+                                debug_image.save(
+                                    f"{utils.debug_dump_dir}/{uid}_sub.png"
+                                )
+
+                                # Screenshot
+                                ss = current_module.screen.copy()
```

Comment:
why not do this before the writing, seems better to first gather all artifacts and then write them

Reply:
Good point, I'll move this up near the other bits!